contrib/win32/win32compat/misc.c

```diff
@@ -1044,58 +1044,62 @@ w32_strerror(int errnum)
 		return _sys_errlist_ext[errnum - EADDRINUSE];
 	return strerror(errnum);
 }
-/* 
- * Temporary implementation of readpassphrase. 
- * TODO - this needs to be reimplemented as per 
- * https://linux.die.net/man/3/readpassphrase
- */
-char * 
-readpassphrase(const char *prompt, char *out, size_t out_len, int flags) {
-	char *askpass = NULL;
-	char *ret = NULL;
 
-	DWORD mode;
-	size_t len = 0;
-	int retr = 0;
+char *
+readpassphrase(const char *prompt, char *outBuf, size_t outBufLen, int flags) {
+	int currentIndex = 0;
+	char ch;
+
+	if (outBufLen == 0) {
+		errno = EINVAL;
+		return(NULL);
+	}
+	
+	while (_kbhit()) _getch();
```

Comment:
Why should we read on a character basis? Why not call an API that reads an entire line (ReadFile on a console handle with console )?
Checkout console APIs on msdn, there are way to control a console handle for echoing, raw mode, etc that could be used for this purpose.